Makefile

```diff
@@ -147,8 +149,8 @@ format: bin/clang-format
 	./tools/symbols.py remove-orphans config/splat.us.stmad.yaml
 patch:
 	$(DIRT_PATCHER) config/dirt.$(VERSION).json
-check: patch
-	sha1sum --check config/check.$(VERSION).sha
+check: config/check.$(VERSION).sha patch $(CHECK_FILES)
```

Comment:
The `$(CHECK_FILES)` is clever, but I do not want to couple `check` with `build` (this is already done by `all`). This is because there are times we do quick checks like `make clean && make build/us/dra.ld && make dra && make check` to check a single overlay instead of having to build everything.

Also I do not want to couple the Makefile task name with the config check name, which already proven to be quite difficult to move things in the past.

Reply:
It seems odd from a build perspective that at the end of that command sequence the shell result would be failure but the human interpretation would be success. Would you be opposed to a separate recipe to ignore do a check while ignoring missing objects (`shasum --check --ignore-missing`) so a workflow like that would still result in success?

> (this is already done by all)

`all` has dependencies on `check` and `build` but make doesn't guarantee an order that dependencies are run (also seen in #739 where the tasks that actually depend on `*_dirs` don't have dependencies but rely on a dependent, which isn't how make is designed). Make only guarantees that dependencies will be completed prior to running the dependent target. Since `check` doesn't explicitly depend on any built artifacts there are often times were `check` will be run (e.g. during a `make -j`) prior to build finishing. I just ran this after a rebase:

```
$> make -j
⋮
./tools/splat_ext/animset.py gen-asm assets/st/wrp/D_80186D88.animset.json build/us/assets/st/wrp/D_80186D88.s
./tools/splat_ext/roomdef.py assets/st/wrp/g_Rooms.roomdef.json build/us/assets/st/wrp/g_Rooms.bin
./tools/splat_ext/layers.py assets/st/wrp/rooms.layers.json build/us/assets/st/wrp/rooms.s
sha1sum --check config/check.us.sha
build/us/main.exe: OK
build/us/DRA.BIN: OK
build/us/WEAPON0.BIN: OK
build/us/RIC.BIN: OK
build/us/CEN.BIN: OK
build/us/F_CEN.BIN: OK
build/us/DRE.BIN: OK
build/us/F_DRE.BIN: OK
build/us/MAD.BIN: OK
build/us/F_MAD.BIN: OK
build/us/NO3.BIN: OK
build/us/F_NO3.BIN: OK
build/us/NP3.BIN: OK
build/us/F_NP3.BIN: OK
build/us/NZ0.BIN: OK
build/us/F_NZ0.BIN: OK
build/us/SEL.BIN: OK
build/us/ST0.BIN: OK
build/us/F_ST0.BIN: OK
build/us/WRP.BIN: OK
build/us/F_WRP.BIN: OK
build/us/RWRP.BIN: OK
build/us/F_RWRP.BIN: OK
build/us/TT_000.BIN: OK
mipsel-linux-gnu-objcopy -O binary build/us/stnz0.elf build/us/NZ0.BIN
mipsel-linux-gnu-ld -r -b binary -o build/us/assets/st/wrp/D_80181250.o build/us/assets/st/wrp/D_80181250.bin
mipsel-linux-gnu-ld -r -b binary -o build/us/assets/st/cen/g_Rooms.o build/us/assets/st/cen/g_Rooms.bin
mipsel-linux-gnu-ld -r -b binary -o build/us/assets/st/wrp/D_80181228.o build/us/assets/st/wrp/D_80181228.bin
mipsel-linux-gnu-ld -r -b binary -o build/us/assets/st/wrp/D_80181278.o build/us/assets/st/wrp/D_80181278.bin
⋮
```

Adding the `CHECK_FILES` dependency ensures the correct ordering and that all of those have completed before `check` runs. 

I'm trying to find a tractable way to correctly model target dependencies so thing something like `make -j dra` would work without the explicit need to clean or require separate make invocations. If those things worked, there would be less cases where a clean is required to begin with and parallel builds have reliable order.

Would you be opposed to an alternate target for `all` that adds the dependency (e.g. `all-check` or `build-check`)?

> Also I do not want to couple the Makefile task name with the config check name, which already proven to be quite difficult to move things in the past.

I'm not sure I follow this. Do you mean you don't want something like a `check-dra` target?

Reply:
My brain must have been drunk when reviewing this. I thought $(CHECK_FILES) would extract target names out of the checksums and expand as `check: dra cen dre ...`, but it actually expands as `check: build/{VERSION}/DRA.BIN ...` . 

I was not aware of `--ignore-missing`.

I am pretty much inexpert on writing `Makefile`s . The sole truth of the current build system working the way it is, is down to some kind of miracle of me trying random things until it worked.

Thank you for the explanation, it makes much more sense where you're heading to with these changes you're proposing in the past few PRs